### ISSUE TITLE
Podcast imported track titles and description do not always match the feed ones. 

### ISSUE THREAD

**Issue by JohnnyC1951** [LibreTime/libretime#431]
This niggle screws up 'track playing' info on some feeds. The simplepie based podcast track availability/imported screen always gets it correct However, the actual imported track can contain something else instead like a random string, track file name or something else - presumably because Libretime gets this info from the ID3 tags? Would it be possible to force the feed's title/description/anything else useful from the RSS feed data, maybe re-tagging the mp3 from the feed on import, using eyed3, id3tag or similar?

**Comment by Robbt** [comment]
Yeah this is indeed an issue. I think it changes the title to the filename (but not always exactly), sometimes it uses the id3 track title. Either way being able to override with the RSS feed info would be a good idea. I could see a couple of ways it could be implemented.

**Comment by JohnnyC1951** [comment]
As this is a big issue here and rather urgent, I am having a go at a simplified simplepie-based proxy that tags the file on the way in.

**Comment by Robbt** [comment]
It might be easier to look at the [podcast ingestion](https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/python_apps/airtime-celery/airtime-celery/tasks.py#L132
) in LibreTime as part of Celery and add it, and then maybe you could submit a PR with the changes.

Here is where we write the [album name to the track](https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/python_apps/airtime-celery/airtime-celery/tasks.py#L180)

It looks like we are just manually passing the podcast name for later insertion to Python, so you'd need to either pass more parameters [here](https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/airtime_mvc/application/services/PodcastEpisodeService.php#L127) in order to do this on the python side of things. I figure that we could also do the track album metadata changes on the Zend PHP side of things as well after they're downloaded. That would most likely entail moving the podcast name functionality out of python as well but it might make more sense than the way I originally coded it if we are going to add this additional functionality.

**Comment by Robbt** [comment]
From what I can tell [updateTrackReference](https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/airtime_mvc/application/services/PodcastEpisodeService.php#L193) would be a good place to work on it on the Zend side of things. But it isn't 100% clear to me when that gets called and whether it has access to SimplePie feed still. There seems to be a decoupling of tracks from their Podcast origin after this point in the code as LibreTime simply doesn't track the origin of a file (only it's metadata from the DB).

**Comment by JohnnyC1951** [comment]
Heh! That's where I got to. ;)  Test proxy is limping using simplexml, not simplepie, but basically working. Just adding housekeeping.

**Comment by Robbt** [comment]
Yeah, one of the challenges in working with LibreTime is that the glue that holds everything together isn't very well documented because none of the original authors of the code are around to explain what they were thinking when they crafted it. Just as well, eventually we might rewrite a fair bit of it.

**Comment by hairmare** [comment]
These also seem to be the bits we got from legacy saas right before we got cut off so it's kinda expected that they are a bit raw.

The title is read using mutagen.easyid3 in a celery task at the moment.

https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/python_apps/airtime-celery/airtime-celery/tasks.py#L158-L165

It would be interesting to know what the difference between the two implementation because this could also be relevant due to our fallback behavior here:

https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/python_apps/airtime-celery/airtime-celery/tasks.py#L189-L194

I think we can fix this by sending more of the simplepie data in the rabbitmq message like we did with the album overriding flag name. Maybe by using a JSON payload containing what we need to allow for customizable album and additionally title and description "overriding" support. We could also fully do away with mutagen and just trust the data from the XML should we go that route.

The code that calls the celery task is here and it's probably reasonably close to the simplepie data. 

https://github.com/LibreTime/libretime/blob/25fbb56689ffe12f0afb80812f0a0c8f1ad48b3a/airtime_mvc/application/services/PodcastEpisodeService.php#L165-L173

Also, maybe the album override feature can help bridge this gap until this is fixed. It's where I remember touching this last in #91.
